### ISSUE TITLE
Possible interpolation issue when using prescribed GLOBAL_OH on different vertical grid

### ISSUE THREAD

**Issue by eastjames** [geoschem/HEMCO#208]
### Name and Institution (Required)

Name: James East
Institution: Harvard (ACMG)

### Confirm you have reviewed the following documentation

- [x] [Support guidelines](https://hemco.readthedocs.io/en/stable/reference/SUPPORT.html)
- [x] [User manuals](https://hemco.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/related-docs.html)
- [x] [Debugging GEOS-Chem and HEMCO errors](https://hemco.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/debug-guide.html)
- [x] [Current and past Github issues](https://github.com/geoschem/HEMCO/issues)

### Description of your issue or question

HEMCO version: 3.6.2
GEOS-Chem version: 14.1.0

#### Background
When running the GEOS-Chem CH4 specialty simulation, I tried using prescribed GLOBAL_OH files that did not originate from GEOS-Chem.

Metadata for these files was edited and conform to Coards requirements. 

Files are monthly time-varying and 3-D, with horizontal and vertical grids different than GEOS-Chem.

When visualizing the files, there aren't obvious errors.

#### Issue
After running GEOS-Chem, plotting methane loss rates to OH show that some of the files lead to apparent errors in the vertical dimension.  It appears that the OH has been collapsed into fewer layers on the GEOS-Chem grid, so OH concentrations are in the wrong layers for the chemistry operations. 

In this plot showing annual average methane loss to OH, note high loss values that show up around 400 hPa. These high values should be occurring at the very top of the model. 
![ncar-cam3 5_lossrate_720](https://user-images.githubusercontent.com/47705516/232153475-5e5984c8-91fc-4b82-bc69-b86fac3fea76.png)

#### More information
There are no errors or warnings in the log files that give any clues. 

When trying multiple files with different vertical and horizontal grids, 2 don't work. The OH files that “work” almost all have more levels than the files that “don’t work”:
```
bad model UM-CAM nlevs = (19,)
bad model NCAR-CAM3.5 nlevs = (26,)
good model CESM-CAM-superfast nlevs = (26,)
good model CICERO-OsloCTM2 nlevs = (60,)
good model CMAM nlevs = (71,)
good model EMAC nlevs = (90,)
good model GEOSCCM nlevs = (72,)
good model GFDL-AM3 nlevs = (48,)
good model GISS-E2-R nlevs = (40,)
good model MIROC-CHEM nlevs = (80,)
good model MOCAGE nlevs = (47,)
```

However, the one of the not-working files (NCAR-CAM3.5) has the nearly same vertical structure as a working file (CESM-CAM-superfast):
![unknown-3_480](https://user-images.githubusercontent.com/47705516/232152304-0d7be807-7b00-4a94-b31c-24da4672564b.png)
![unknown-4_480](https://user-images.githubusercontent.com/47705516/232152328-09674a74-9b44-491f-afc1-adefaaee1165.png)

Despite appearing similar, the actual values between these two files are slightly different:
```
(base) [jeast@holy7c24403 oh]$ ncdump -v lev ACCMIP/vmroh_ACCMIP-monthly_NCAR-CAM3.5_acchist_r1i1p1_200901-200912.gcclassic.nc | tail -n 10
 lev = 0.992556154727936, 0.970554828643799, 0.92964893579483, 
    0.867160797119141, 0.787702143192291, 0.696796298027039, 
    0.600524187088013, 0.510455310344696, 0.433895260095596, 
    0.368818014860153, 0.313501268625259, 0.266481161117554, 
    0.226513266563416, 0.192539945244789, 0.163662075996399, 
    0.139115408062935, 0.118250340223312, 0.100514702498913, 
    0.0854391232132912, 0.0700591504573822, 0.0531146079301834, 
    0.0372302904725075, 0.0239446274936199, 0.0139672141522169, 
    0.00738881388679147, 0.00354463816620409 ;
}
(base) [jeast@holy7c24403 oh]$ ncdump -v lev ACCMIP/vmroh_ACCMIP-monthly_CESM-CAM-superfast_acchist_r1i1p1_200901-200912.gcclassic.nc | tail -n 10
data:

 lev = 0.9925561, 0.970554830000001, 0.929648875000002, 0.867160760000001, 
    0.787702060000003, 0.696796290000003, 0.600524200000003, 
    0.510455255000002, 0.433895225000001, 0.368817980000002, 
    0.313501265000001, 0.266481155, 0.226513265, 0.192539935, 0.16366207, 
    0.139115395, 0.118250335, 0.100514695, 0.0854391150000003, 
    0.0700591500000003, 0.0531146050000002, 0.0372302900000001, 0.023944625, 
    0.013967214, 0.00738881350000001, 0.00354463800000001 ;
}
```

**Comment by jimmielin** [comment]
Hi James, all,

I am interested in this as well since it's a possible issue with the assumptions in vertical interpolation specific to GC-Classic. The relevant code is in HEMCO, `src/Core/hco_interp_mod.F90` in the `ModelLev_Interpolate` routine.

The comments say (and might be outdated) that the following combinations will be collapsed or inflated onto the model levels **regardless of what the coordinates in the files say** - only the # of levels is read. I believe you may be running into this issue.

```
! \item Native  GEOS-5 onto reduced GEOS-5 (72 --> 47 levels)
! \item Reduced GEOS-5 onto native  GEOS-5 (47 --> 72 levels)
! \item Native  GEOS-4 onto reduced GEOS-4 (55 --> 30 levels)
! \item Reduced GEOS-4 onto native  GEOS-4 (30 --> 55 levels)
! \item Native  GEOS-5 onto native  GEOS-4 (72 --> 55 levels)
! \item Reduced GEOS-5 onto native  GEOS-4 (47 --> 55 levels)
! \item Native  GEOS-5 onto reduced GEOS-4 (72 --> 30 levels)
! \item Reduced GEOS-5 onto reduced GEOS-4 (47 --> 30 levels)
```

This only happens in GEOS-Chem classic because `ModelLev_Interpolate` is only called in `REGRID_MAPA2A` which is only called in GEOS-Chem classic. Other models (`MODEL_WRF` and `MODEL_CESM`) enforce MESSy regridding and `IsModelLevel = .false.` bypassing any of this behavior, and MAPL does its own regridding independent of what is described in `hcoio_read_std_mod.F90`.

Haipeng

**Comment by eastjames** [comment]
Hi all,

If I regrid the prescribed GLOBAL_OH fields offline before running GEOS-Chem, the loss rates look as expected, suggesting that the vertical regridding and/or layer collapsing suggested by @jimmielin is a likely cause.

![NCAR-CAM3 5_lossrate_fix](https://github.com/geoschem/HEMCO/assets/47705516/9f345c45-65fb-42f6-bb2d-f4a5c9dc4f51)

Thanks,
James

**Comment by msulprizio** [comment]
Thanks for following up @eastjames. I've marked this as a bug, but we may not have time to look into this for a while.

**Comment by yantosca** [comment]
The vertical interpolation may be fixed by PR #235, which has been merged into the GEOS-Chem 14.2.1 / HEMCO 3.7.1 development stream.

**Comment by yantosca** [comment]
We can close this issue as PR #235 has been merged in to HEMCO 3.7.1.
